<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.6"/>
<title>libsequence: Sequence::PolySNP Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">libsequence
   &#160;<span id="projectnumber">1.8.8</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.6 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="examples.html"><span>Examples</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('classSequence_1_1PolySNP.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pro-methods">Protected Member Functions</a> &#124;
<a href="#pro-attribs">Protected Attributes</a> &#124;
<a href="classSequence_1_1PolySNP-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">Sequence::PolySNP Class Reference<div class="ingroups"><a class="el" href="group__popgenanalysis.html">Analysis of molecular population genetic data</a></div></div>  </div>
</div><!--header-->
<div class="contents">

<p>Molecular population genetic analysis.  
 <a href="classSequence_1_1PolySNP.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="PolySNP_8hpp_source.html">Sequence/PolySNP.hpp</a>&gt;</code></p>
<div class="dynheader">
Inheritance diagram for Sequence::PolySNP:</div>
<div class="dyncontent">
 <div class="center">
  <img src="classSequence_1_1PolySNP.png" usemap="#Sequence::PolySNP_map" alt=""/>
  <map id="Sequence::PolySNP_map" name="Sequence::PolySNP_map">
<area href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. " alt="Sequence::PolySIM" shape="rect" coords="0,56,124,80"/>
</map>
 </div></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:afa3702f4a95a35c3b5fac9d84370b017"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#afa3702f4a95a35c3b5fac9d84370b017">PolySNP</a> (const <a class="el" href="classSequence_1_1PolyTable.html">Sequence::PolyTable</a> *data, bool haveOutgroup=false, unsigned outgroup=0, bool totMuts=true)</td></tr>
<tr class="separator:afa3702f4a95a35c3b5fac9d84370b017"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac96be71be8e14862df2c8f232a19b1b6"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac96be71be8e14862df2c8f232a19b1b6"></a>
&#160;</td><td class="memItemRight" valign="bottom"><b>PolySNP</b> (const <a class="el" href="classSequence_1_1PolySNP.html">PolySNP</a> &amp;)=delete</td></tr>
<tr class="separator:ac96be71be8e14862df2c8f232a19b1b6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a421c381faaa312ee410063b1b132de48"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a421c381faaa312ee410063b1b132de48"></a>
<a class="el" href="classSequence_1_1PolySNP.html">PolySNP</a> &amp;&#160;</td><td class="memItemRight" valign="bottom"><b>operator=</b> (const <a class="el" href="classSequence_1_1PolySNP.html">PolySNP</a> &amp;)=delete</td></tr>
<tr class="separator:a421c381faaa312ee410063b1b132de48"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9c26d27c7b0aaf1faf859272871b3cf5"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a9c26d27c7b0aaf1faf859272871b3cf5">ThetaPi</a> (void) const </td></tr>
<tr class="separator:a9c26d27c7b0aaf1faf859272871b3cf5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a942cec5aec3bd820c9029609969d4b57"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a942cec5aec3bd820c9029609969d4b57">ThetaW</a> (void) const </td></tr>
<tr class="separator:a942cec5aec3bd820c9029609969d4b57"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4ec4a78eaa7f3f2bac0a9121fd466d0a"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a4ec4a78eaa7f3f2bac0a9121fd466d0a">ThetaH</a> (void) const </td></tr>
<tr class="separator:a4ec4a78eaa7f3f2bac0a9121fd466d0a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab6a3004d3fae4afb050158deb17178b9"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#ab6a3004d3fae4afb050158deb17178b9">ThetaL</a> (void) const </td></tr>
<tr class="separator:ab6a3004d3fae4afb050158deb17178b9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa5614fba76be5730e4618032820a9b0c"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#aa5614fba76be5730e4618032820a9b0c">VarPi</a> (void) const </td></tr>
<tr class="separator:aa5614fba76be5730e4618032820a9b0c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a22535bbb53cf745498c7c84b774319e3"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a22535bbb53cf745498c7c84b774319e3">StochasticVarPi</a> (void) const </td></tr>
<tr class="separator:a22535bbb53cf745498c7c84b774319e3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a816138ea5c26ba2923a60b128d54f4e1"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a816138ea5c26ba2923a60b128d54f4e1">SamplingVarPi</a> (void) const </td></tr>
<tr class="separator:a816138ea5c26ba2923a60b128d54f4e1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05276ed08aa7184738228045dd87ec15"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a05276ed08aa7184738228045dd87ec15">VarThetaW</a> (void) const </td></tr>
<tr class="separator:a05276ed08aa7184738228045dd87ec15"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a582904af09c17b0644b1baa3bc3366d5"><td class="memItemLeft" align="right" valign="top">unsigned&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a582904af09c17b0644b1baa3bc3366d5">NumPoly</a> (void) const </td></tr>
<tr class="separator:a582904af09c17b0644b1baa3bc3366d5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1789e4231da418c355f49bdbd67941a3"><td class="memItemLeft" align="right" valign="top">virtual unsigned&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a1789e4231da418c355f49bdbd67941a3">NumMutations</a> (void) const </td></tr>
<tr class="separator:a1789e4231da418c355f49bdbd67941a3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad35151bb09bb49e740515f64ca97c754"><td class="memItemLeft" align="right" valign="top">virtual unsigned&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#ad35151bb09bb49e740515f64ca97c754">NumSingletons</a> (void) const </td></tr>
<tr class="separator:ad35151bb09bb49e740515f64ca97c754"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad634de63223e2d8ad3f27e28cbf3fff3"><td class="memItemLeft" align="right" valign="top">virtual unsigned&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#ad634de63223e2d8ad3f27e28cbf3fff3">NumExternalMutations</a> (void) const </td></tr>
<tr class="separator:ad634de63223e2d8ad3f27e28cbf3fff3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad236330defb0426b61265d3b955c6bef"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#ad236330defb0426b61265d3b955c6bef">TajimasD</a> (void) const </td></tr>
<tr class="separator:ad236330defb0426b61265d3b955c6bef"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a619fa8633b37008bca10bacc018ee628"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a619fa8633b37008bca10bacc018ee628">Hprime</a> (const bool &amp;likeThorntonAndolfatto=false) const </td></tr>
<tr class="separator:a619fa8633b37008bca10bacc018ee628"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2c28e0c7330d05bc8fec1039eb64abe7"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a2c28e0c7330d05bc8fec1039eb64abe7">Dnominator</a> (void) const </td></tr>
<tr class="separator:a2c28e0c7330d05bc8fec1039eb64abe7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0c31e0d79d1aff7c51349b1ae23052c0"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a0c31e0d79d1aff7c51349b1ae23052c0">FuLiD</a> (void) const </td></tr>
<tr class="separator:a0c31e0d79d1aff7c51349b1ae23052c0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a011c4369b8f38d26fe5702dc8881bd4c"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a011c4369b8f38d26fe5702dc8881bd4c">FuLiF</a> (void) const </td></tr>
<tr class="separator:a011c4369b8f38d26fe5702dc8881bd4c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2b71875afc4f53de1ed5338e3a70c29f"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a2b71875afc4f53de1ed5338e3a70c29f">FuLiDStar</a> (void) const </td></tr>
<tr class="separator:a2b71875afc4f53de1ed5338e3a70c29f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4e102794535af0a7bcc294cc958f3358"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a4e102794535af0a7bcc294cc958f3358">FuLiFStar</a> (void) const </td></tr>
<tr class="separator:a4e102794535af0a7bcc294cc958f3358"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5cb1e8e0583eaf95f8baf5d4383e8cca"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a5cb1e8e0583eaf95f8baf5d4383e8cca">DandVH</a> (void) const </td></tr>
<tr class="separator:a5cb1e8e0583eaf95f8baf5d4383e8cca"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1be376ce9d69d7bc0016fac306124454"><td class="memItemLeft" align="right" valign="top">unsigned&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a1be376ce9d69d7bc0016fac306124454">DandVK</a> (void) const </td></tr>
<tr class="separator:a1be376ce9d69d7bc0016fac306124454"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a00d617041ba15a1593ffd85134a6a315"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a00d617041ba15a1593ffd85134a6a315">WallsB</a> (void) const </td></tr>
<tr class="separator:a00d617041ba15a1593ffd85134a6a315"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa2360fb83235da3d1ff16eec11454b99"><td class="memItemLeft" align="right" valign="top">virtual unsigned&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#aa2360fb83235da3d1ff16eec11454b99">WallsBprime</a> (void) const </td></tr>
<tr class="separator:aa2360fb83235da3d1ff16eec11454b99"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a016a664f6cf9380310910d86482b1f7e"><td class="memItemLeft" align="right" valign="top">virtual double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a016a664f6cf9380310910d86482b1f7e">WallsQ</a> (void) const </td></tr>
<tr class="separator:a016a664f6cf9380310910d86482b1f7e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afaaf1b0dc8b931eaaa7a801a4db12229"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#afaaf1b0dc8b931eaaa7a801a4db12229">HudsonsC</a> (void) const </td></tr>
<tr class="separator:afaaf1b0dc8b931eaaa7a801a4db12229"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a16f0711c46a6b693cdb20b0956416585"><td class="memItemLeft" align="right" valign="top">virtual unsigned&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a16f0711c46a6b693cdb20b0956416585">Minrec</a> (void) const </td></tr>
<tr class="separator:a16f0711c46a6b693cdb20b0956416585"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a50eea85970a1e0a4eb3a049ded42b10d"><td class="memItemLeft" align="right" valign="top">std::vector&lt; std::vector<br class="typebreak"/>
&lt; double &gt; &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a50eea85970a1e0a4eb3a049ded42b10d">Disequilibrium</a> (const unsigned &amp;mincount=1, const double &amp;max_marker_distance=std::numeric_limits&lt; double &gt;::max()) const </td></tr>
<tr class="separator:a50eea85970a1e0a4eb3a049ded42b10d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Protected Member Functions</h2></td></tr>
<tr class="memitem:af89d438b6bb1df23dac3b2162e42d260"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#af89d438b6bb1df23dac3b2162e42d260">DepaulisVeuilleStatistics</a> (void) const </td></tr>
<tr class="separator:af89d438b6bb1df23dac3b2162e42d260"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a47a48b2aa71f1ff5beb9f44f384ae47a"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a47a48b2aa71f1ff5beb9f44f384ae47a"></a>
virtual void&#160;</td><td class="memItemRight" valign="bottom"><b>WallStats</b> (void) const </td></tr>
<tr class="separator:a47a48b2aa71f1ff5beb9f44f384ae47a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac2e80d4fe8be5231aa99993949c123e7"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#ac2e80d4fe8be5231aa99993949c123e7">a_sub_n</a> (void) const </td></tr>
<tr class="separator:ac2e80d4fe8be5231aa99993949c123e7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac930e586b07a9b64ce3c4c53c8b3d401"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#ac930e586b07a9b64ce3c4c53c8b3d401">a_sub_n_plus1</a> (void) const </td></tr>
<tr class="separator:ac930e586b07a9b64ce3c4c53c8b3d401"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0f4687d987b2548be37d902321b7d9aa"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a0f4687d987b2548be37d902321b7d9aa">b_sub_n</a> (void) const </td></tr>
<tr class="separator:a0f4687d987b2548be37d902321b7d9aa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:add59a3ecc2737306b4856046b86a0581"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#add59a3ecc2737306b4856046b86a0581">b_sub_n_plus1</a> (void) const </td></tr>
<tr class="separator:add59a3ecc2737306b4856046b86a0581"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a480d29911ebea7edfaba976f47035d76"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a480d29911ebea7edfaba976f47035d76">c_sub_n</a> (void) const </td></tr>
<tr class="separator:a480d29911ebea7edfaba976f47035d76"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a383727692316e4e110010b2429f6f8f1"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSequence_1_1PolySNP.html#a383727692316e4e110010b2429f6f8f1">d_sub_n</a> (void) const </td></tr>
<tr class="separator:a383727692316e4e110010b2429f6f8f1"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-attribs"></a>
Protected Attributes</h2></td></tr>
<tr class="memitem:a927043714e78254f032ce6581b04ea12"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a927043714e78254f032ce6581b04ea12"></a>
std::unique_ptr&lt; <a class="el" href="structSequence_1_1__PolySNPImpl.html">_PolySNPImpl</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><b>rep</b></td></tr>
<tr class="separator:a927043714e78254f032ce6581b04ea12"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Molecular population genetic analysis. </p>
<p>Example Use:<br/>
<br/>
</p>
<div class="fragment"><div class="line"><span class="preprocessor">#include &lt;iostream&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;vector&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="PolySNP_8hpp.html">Sequence/PolySNP.hpp</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="Fasta_8hpp.html">Sequence/Fasta.hpp</a>&gt;</span></div>
<div class="line"><span class="preprocessor">#include &lt;<a class="code" href="PolySites_8hpp.html">Sequence/PolySites.hpp</a>&gt;</span></div>
<div class="line"><span class="keyword">using namespace </span>std;</div>
<div class="line"><span class="keyword">using namespace </span>Sequence;</div>
<div class="line"><span class="keywordtype">int</span> main</div>
<div class="line">{</div>
<div class="line">  vector&lt;Fasta&gt; data;</div>
<div class="line">  <a class="code" href="namespaceSequence_1_1Alignment.html#a27f1d75d0ac948dab690542f574b792e">Alignment::GetData</a>(data,<span class="stringliteral">&quot;popdata.fasta&quot;</span>);</div>
<div class="line">  assert(<a class="code" href="namespaceSequence_1_1Alignment.html#a5ab64478af8efead7e80ada3722d0342">Alignment::IsAlignment</a>(data))</div>
<div class="line">  if (Alignment::<a class="code" href="namespaceSequence.html#a6b8d03d21bed077f2f95790a2bc55a3a">Gapped</a>(data))</div>
<div class="line">  {</div>
<div class="line">    <a class="code" href="namespaceSequence_1_1Alignment.html#a873d7abdf6bf1a915b20f48c29cc60ce">Alignment::RemoveTerminalGaps</a>(data);</div>
<div class="line">  }</div>
<div class="line">  <a class="code" href="classSequence_1_1PolySites.html">PolySites</a> *polytable = <span class="keyword">new</span> <a class="code" href="classSequence_1_1PolySites.html">PolySites</a>(data);</div>
<div class="line">  <a class="code" href="classSequence_1_1PolySNP.html">PolySNP</a> *analyze = <span class="keyword">new</span> <a class="code" href="classSequence_1_1PolySNP.html#afa3702f4a95a35c3b5fac9d84370b017">PolySNP</a>(data,<span class="keyword">false</span>,0);</div>
<div class="line">  cout &lt;&lt; <span class="stringliteral">&quot;Tajima&#39;s D is &quot;</span> &lt;&lt; analyze-&gt;<a class="code" href="classSequence_1_1PolySNP.html#ad236330defb0426b61265d3b955c6bef">TajimasD</a>() &lt;&lt; endl;</div>
<div class="line">  <span class="keyword">delete</span> polytable;</div>
<div class="line">  <span class="keyword">delete</span> analyze;</div>
<div class="line">  exit(1);</div>
<div class="line">}</div>
</div><!-- fragment --> <dl class="section warning"><dt>Warning</dt><dd>The routines to calculate nucleotide diversity, numbers of singletons, etc. all handle missing data (the <code>N</code> character). However, all summary statistics involved in "tests of 
neutrality" are, strictly speaking, undefined if missing data are present. The reason for this is because the denominators of the statistics are functions of the sample sizes, and no explicit formulae exist when the sample size varies from site to site (which is the case when there are missing data). In short, if you want to be rigorous, you can only really count up nucleotide diversity and a few other statistics if your data contain untyped SNPs. However, the routines present in <code>libsequence</code> will happily go and calculate the summary statistics for you, and it is up to you to be aware that you are writing a program that may give biased results. To date, the magnitude and direction of the bias remains unknown. Functions (and hence the statistics) that are affected have warnings in their documentation. </dd></dl>
<dl class="section note"><dt>Note</dt><dd>As of libsequence 1.4.1, the routines in this class explicity check for gaps in the polymorphism table. This provides an additional safeguard for cases where <a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a> objects are created and sites with gaps are left in. </dd></dl>
<dl><dt><b>Examples: </b></dt><dd><a class="el" href="slidingWindow_8cc-example.html#_a5">slidingWindow.cc</a>, and <a class="el" href="slidingWindow2_8cc-example.html#_a4">slidingWindow2.cc</a>.</dd>
</dl>
<p>Definition at line <a class="el" href="PolySNP_8hpp_source.html#l00083">83</a> of file <a class="el" href="PolySNP_8hpp_source.html">PolySNP.hpp</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="afa3702f4a95a35c3b5fac9d84370b017"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Sequence::PolySNP::PolySNP </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classSequence_1_1PolyTable.html">Sequence::PolyTable</a> *&#160;</td>
          <td class="paramname"><em>data</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>haveOutgroup</em> = <code>false</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">unsigned&#160;</td>
          <td class="paramname"><em>outgroup</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>totMuts</em> = <code>true</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">explicit</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">data</td><td>a valid object of type <a class="el" href="classSequence_1_1PolyTable.html" title="The base class for polymorphism tables. ">Sequence::PolyTable</a> </td></tr>
    <tr><td class="paramname">haveOutgroup</td><td><code>true</code> if an outgroup is present, <code>false</code> otherwise </td></tr>
    <tr><td class="paramname">outgroup</td><td>if <em>haveOutgroup</em> is <code>true</code>, <em>outgroup</em> is the index of that sequence in data </td></tr>
    <tr><td class="paramname">totMuts</td><td>if true (the default) use the total number of inferred mutations, otherwise use the total number of polymorphic sites in calculations </td></tr>
  </table>
  </dd>
</dl>
<dl class="section note"><dt>Note</dt><dd>this constructor allocates a pointer to an implementation class, which automatically pre-processes the SNP data </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00163">163</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="ac2e80d4fe8be5231aa99993949c123e7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::a_sub_n </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p class="formulaDsp">
<img class="formulaDsp" alt="\[a_n=\sum_{i=1}^{i=n-1}\frac{1}{i}.\ \]" src="form_31.png"/>
</p>
<p> This is the denominator of Watterson's Theta (see <a class="el" href="classSequence_1_1PolySNP.html#a942cec5aec3bd820c9029609969d4b57">PolySNP::ThetaW</a>) </p>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01157">1157</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="ac930e586b07a9b64ce3c4c53c8b3d401"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::a_sub_n_plus1 </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p class="formulaDsp">
<img class="formulaDsp" alt="\[a_{n+1}=\sum_{i=1}^{i=n}\frac{1}{i}\ \]" src="form_32.png"/>
</p>
 <dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01173">1173</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a0f4687d987b2548be37d902321b7d9aa"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::b_sub_n </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p class="formulaDsp">
<img class="formulaDsp" alt="\[b_n=\sum_{i=1}^{i=n-1}\frac{1}{i^2}\ \]" src="form_33.png"/>
</p>
 <dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01190">1190</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="add59a3ecc2737306b4856046b86a0581"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::b_sub_n_plus1 </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p class="formulaDsp">
<img class="formulaDsp" alt="\[b_n=\sum_{i=1}^{i=n}\frac{1}{i^2}\ \]" src="form_34.png"/>
</p>
 <dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>
<dl class="section author"><dt>Author</dt><dd>Joshua Shapiro </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01205">1205</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a480d29911ebea7edfaba976f47035d76"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::c_sub_n </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p class="formulaDsp">
<img class="formulaDsp" alt="\[ c_n=\left\{\begin{array}{cl} 1 , &amp; when\ n = 2 \\ \frac{2 \times (n \times a_n - 2 \times (n-1))}{(n-1) \times (n-2)}, &amp; when\ n > 2 \\ \end{array}\right.\ \]" src="form_35.png"/>
</p>
 <dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01221">1221</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a383727692316e4e110010b2429f6f8f1"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::d_sub_n </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p class="formulaDsp">
<img class="formulaDsp" alt="\[\ d_n=\frac{2}{n-1} \times (1.5 - \frac{2 \times a_{n+1}}{n-2} - \frac{1}{n})\ \]" src="form_36.png"/>
</p>
 <dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01247">1247</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a5cb1e8e0583eaf95f8baf5d4383e8cca"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::DandVH </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>To check if two sequences are unique, Sequence::Comparisons::Different is used, which does not allow missing data to result in 2 sequences being considered different (as they would be if you simply used thestd::string comparison operators == or !=) </p>
<dl class="section return"><dt>Returns</dt><dd>the haplotype diversity of the data. </dd></dl>
<dl><dt><b>Examples: </b></dt><dd><a class="el" href="bottleneck_8cc-example.html#a11">bottleneck.cc</a>.</dd>
</dl>
<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01264">1264</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a1be376ce9d69d7bc0016fac306124454"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">unsigned Sequence::PolySNP::DandVK </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>To check if two sequences are unique, Sequence::Comparisons::Different is used, which does not allow missing data to result in 2 sequences being considered different (as they would be if you simply used the std::string comparison operators == or !=) </p>
<dl class="section return"><dt>Returns</dt><dd>number of haplotypes in the sample </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01281">1281</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="af89d438b6bb1df23dac3b2162e42d260"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void Sequence::PolySNP::DepaulisVeuilleStatistics </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Calculate the number of haplotypes in the sample, and haplotype diversity. Unlike Depaulis and Veuille's original paper, this routine uses an unbiased calculation of haplotype diversity (i.e. divide by n choose 2). <br/>
To check if two sequences are unique, Sequence::Comparisons::Different is used, which does not allow missing data to result in 2 sequences being considered different (as they would be if you simply used the std::string comparison operators == or !=) </p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00757">757</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a50eea85970a1e0a4eb3a049ded42b10d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt; std::vector&lt; double &gt; &gt; Sequence::PolySNP::Disequilibrium </td>
          <td>(</td>
          <td class="paramtype">const unsigned &amp;&#160;</td>
          <td class="paramname"><em>mincount</em> = <code>1</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const double &amp;&#160;</td>
          <td class="paramname"><em>max_marker_distance</em> = <code>std::numeric_limits&lt;double&gt;::max()</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>A vector of statistics related to LD and distance in the sample. An empty vector is returned if there are &lt; 2 polymorphic sites in the sample. See the documentation for Recombination::Disequilibrium for a description of the return vector. </dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">mincount</td><td>a frequency filter. A polymorphism must be present at least <em>mincount</em> times in the data </td></tr>
  </table>
  </dd>
</dl>
<dl class="section note"><dt>Note</dt><dd>For D and D', the 11 gamete is defined as follows: If no outgroup is present, it refers to the genotype of minor alleles at both sites. If there is an outgroup, it is based on the genotype of derived alleles at both sites. </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01427">1427</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a2c28e0c7330d05bc8fec1039eb64abe7"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::Dnominator </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>
<dl class="section return"><dt>Returns</dt><dd>Denominator of Tajima's D, or nan if there are no polymorphic sites </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a8f8b95b3331ac33c19e9b9cec4b5e23e">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00723">723</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a0c31e0d79d1aff7c51349b1ae23052c0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::FuLiD </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>The Fu and Li (1993) D statistic, or nan if there are no polymorphic sites. </dd></dl>
<dl class="section note"><dt>Note</dt><dd>For sequence data, an outgroup is required. This requirement is checked by assert() </dd></dl>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#ae185098e0ce4ae373601a61d6175486b">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01021">1021</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a2b71875afc4f53de1ed5338e3a70c29f"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::FuLiDStar </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>
<dl class="section return"><dt>Returns</dt><dd>Fu and Li (1993) D*, or nan if there are no polymorphic sites </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a06911f2e977e72104eb16075157cb101">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01084">1084</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a011c4369b8f38d26fe5702dc8881bd4c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::FuLiF </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>Fu and Li (1993) F statistic, or nan if there are no polymorphic sites </dd></dl>
<dl class="section note"><dt>Note</dt><dd>For sequence data, an outgroup is required, else undefined </dd></dl>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a2aaaaf19c345bab6cdb9e8b3ed048f04">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01049">1049</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a4e102794535af0a7bcc294cc958f3358"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::FuLiFStar </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Fu and Li (1993) F* statistic. Incorporates correction from Simonsen et al. (1995) Genetics 141: 413, eqn A5. </p>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>
<dl class="section return"><dt>Returns</dt><dd>Fu and Li (1993) F* statistic, or nan if there are no polymorphic sites </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#ad9db7aeb8cfb4ff579bb87b0dd73fd5c">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01117">1117</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a619fa8633b37008bca10bacc018ee628"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::Hprime </td>
          <td>(</td>
          <td class="paramtype">const bool &amp;&#160;</td>
          <td class="paramname"><em>likeThorntonAndolfatto</em> = <code>false</code></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>ThetaPi-ThetaH/(~Var(ThetaPi-ThetaH)). This corresponds to Equation 5 in Thornton and Andolfatto (Genetics) "Approximate Bayesian Inference reveals evidence 
for a recent, severe, bottleneck in a Netherlands population of Drosophila melanogaster" and Equation 13 of Zeng et al. (2006) Genetics 1431-1439 </dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">likeThorntonAndolfatto</td><td>The calculation of H' requires calculation of <img class="formulaInl" alt="$\theta^2$" src="form_28.png"/>. In Thornton and Andolfatto, we simply used <img class="formulaInl" alt="$\widehat\theta_W^2$" src="form_29.png"/>, which is slightly biased. By default, this function calculates <img class="formulaInl" alt="$\theta^2=\frac{S(S-1)}{a_n^2+b_n}$" src="form_30.png"/>, unless this bool is set to false, in which case <img class="formulaInl" alt="$\widehat\theta_W^2$" src="form_29.png"/> is used. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section note"><dt>Note</dt><dd>returns nan if there are 0 polymorphic sites </dd></dl>
<dl class="section author"><dt>Author</dt><dd>Joshua Shapiro </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#ade50b9770220227efa863e4d47c727d5">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00674">674</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="afaaf1b0dc8b931eaaa7a801a4db12229"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::HudsonsC </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>Hudson's (1987) estimator of <img class="formulaInl" alt="$\rho=4Nc$" src="form_37.png"/>, an estimator of the population recombination rate that depends on the variance of the site frequencies. The calculation is made by a call to <a class="el" href="namespaceSequence_1_1Recombination.html#a69224c56b3ca7cfe9a28685217e02a08">Recombination::HudsonsC</a> </dd></dl>
<dl class="section note"><dt>Note</dt><dd>Will return nan if there are no polymorphic sites </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01298">1298</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a16f0711c46a6b693cdb20b0956416585"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">unsigned Sequence::PolySNP::Minrec </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>The minimum number of recombination events observed in the sample (Hudson and Kaplan 1985). Will return SEQMAXUNSIGNED if there are &lt; 2 segregating sites. </dd></dl>
<dl class="section note"><dt>Note</dt><dd>Code is a modification of that provided by Jeff Wall </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a11ef31f008a3619deb28ccfee32832d6">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01314">1314</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="ad634de63223e2d8ad3f27e28cbf3fff3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">unsigned Sequence::PolySNP::NumExternalMutations </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the number of derived singletons. </dd></dl>
<dl class="section note"><dt>Note</dt><dd>For sequence data, an outgroup is required. Will return SEQMAXUNSIGNED if that is not the case. </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#ae40261c55f9ffffdeed02c9336ebb64e">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00622">622</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a1789e4231da418c355f49bdbd67941a3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">unsigned Sequence::PolySNP::NumMutations </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the total number of mutations in the data. The number of mutations per site = number of states per site - 1 </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#af156389d7b2d1c7b8b91a7823a6a11ca">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00570">570</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a582904af09c17b0644b1baa3bc3366d5"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">unsigned Sequence::PolySNP::NumPoly </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>the number of polymorphic (segregating) sites in data </dd></dl>
<dl><dt><b>Examples: </b></dt><dd><a class="el" href="msstats_8cc-example.html#a4">msstats.cc</a>.</dd>
</dl>
<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00553">553</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="ad35151bb09bb49e740515f64ca97c754"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">unsigned Sequence::PolySNP::NumSingletons </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>number of polymorphisms that appear once in the data, without respect to ancestral/derived </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a9467d836e65e2a81f90b6a8da4e81bd2">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00589">589</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a816138ea5c26ba2923a60b128d54f4e1"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::SamplingVarPi </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Component of variance of mean pairwise differences from sampling. Tajima in Takahata/Clark book, (15) </p>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00986">986</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a22535bbb53cf745498c7c84b774319e3"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::StochasticVarPi </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Stochastic variance of mean pairwise differences. Tajima in Takahata/Clark book, (14). </p>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00971">971</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="ad236330defb0426b61265d3b955c6bef"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::TajimasD </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>A common summary of the site frequency spectrum. Proportional to <img class="formulaInl" alt="$\widehat\theta_\pi-\widehat\theta_W$" src="form_27.png"/>. This routine does calculate the denominator of the test statistic. </p>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>
<dl class="section return"><dt>Returns</dt><dd>Tajima's D, or nan if there are no polymorphic sites </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a9b1af310ffc68287837f2c5ef039ee9b">Sequence::PolySIM</a>.</p>
<dl><dt><b>Examples: </b></dt><dd><a class="el" href="slidingWindow_8cc-example.html#a6">slidingWindow.cc</a>, and <a class="el" href="slidingWindow2_8cc-example.html#a5">slidingWindow2.cc</a>.</dd>
</dl>
<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00653">653</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a4ec4a78eaa7f3f2bac0a9121fd466d0a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::ThetaH </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Calculate Theta ( = 4Nu) from site homozygosity, a la Fay and Wu (2000). This statistic is problematic in general to calculate when there are multiple hits. The test requires that the ancestral state (inferred from the outgroup) still be segregating in the ingroup. If that is not true, the site is skipped.<br/>
 If there are &gt;= 2 derived states inferred, a "missing data" approach is taken.<br/>
For example:<br/>
Outgroup : <br/>
A<br/>
Ingroup : <br/>
A<br/>
A<br/>
A<br/>
G<br/>
G<br/>
T<br/>
Gets treated as two sites:<br/>
A A<br/>
A A<br/>
A A<br/>
G N<br/>
G N<br/>
N T<br/>
<br/>
This keeps the expectation of the statistic equal to <img class="formulaInl" alt="$\theta$" src="form_0.png"/>, and uses the correct number of derived mutations ovserved in the data. </p>
<dl class="section note"><dt>Note</dt><dd>When using <a class="el" href="classSequence_1_1PolySNP.html" title="Molecular population genetic analysis. ">PolySNP</a>, an outgroup is required. When using <a class="el" href="classSequence_1_1PolySIM.html" title="Analysis of coalescent simulation data. ">PolySIM</a>, which is constructed with a <a class="el" href="classSequence_1_1SimData.html" title="Data from coalescent simulations. ">SimData</a> *, an outgroup is not required (as the 0,1 coding of the data refers to ancestral and derived, respectively). </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a0aa19c3cdfddc2bc5b04142b85ab2499">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00298">298</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="ab6a3004d3fae4afb050158deb17178b9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::ThetaL </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Calculate Theta ( = 4Nu) from site homozygosity, corresponding to equation 1 in Thornton and Andolfatto (Genetics) "Approximate Bayesian Inference reveals evidence 
for a recent, severe, bottleneck in a Netherlands population of Drosophila melanogaster," (although we labelled in <img class="formulaInl" alt="$theta_\eta$" src="form_26.png"/> in that paper) The test requires that the ancestral state (inferred from the outgroup) still be segregating in the ingroup. If that is not true, the site is skipped.<br/>
 If there are &gt;= 2 derived states inferred, a "missing data" approach is taken.<br/>
For example:<br/>
Outgroup : <br/>
A<br/>
Ingroup : <br/>
A<br/>
A<br/>
A<br/>
G<br/>
G<br/>
T<br/>
Gets treated as two sites:<br/>
A A<br/>
A A<br/>
A A<br/>
G N<br/>
G N<br/>
N T<br/>
<br/>
This keeps the expectation of the statistic equal to <img class="formulaInl" alt="$\theta$" src="form_0.png"/>, and uses the correct number of derived mutations ovserved in the data. </p>
<dl class="section note"><dt>Note</dt><dd>For sequence data, an outgroup is required. This requirement is checked by assert() </dd></dl>
<dl class="section author"><dt>Author</dt><dd>Joshua Shapiro </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#adf157e278c3b0d2cc61d2e106bec01f2">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00429">429</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a9c26d27c7b0aaf1faf859272871b3cf5"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::ThetaPi </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Calculated here as the sum of 1.0 - sum of site homozygosity accross sites.<br/>
 </p>
<p class="formulaDsp">
<img class="formulaDsp" alt="\[ \widehat\theta_\pi=\sum_{i=1}^{i=S}(1-\sum_{j=1}^{j=4}\frac{k_{j,i} \times (k_{j,i}-1)}{n_i \times (n_i - 1)});k_{j,i}>0 \]" src="form_17.png"/>
</p>
<p> Where <img class="formulaInl" alt="$S$" src="form_18.png"/> is the number of segregating sites, <img class="formulaInl" alt="$k_{j,i}$" src="form_19.png"/> is the number of occurences of the <img class="formulaInl" alt="$j_{th}$" src="form_20.png"/> character state at site <img class="formulaInl" alt="$i$" src="form_21.png"/>, and <img class="formulaInl" alt="$n_i$" src="form_22.png"/> is the sample size at site <img class="formulaInl" alt="$i$" src="form_21.png"/>. Calculating the statistic in this manner makes it easy to generalize to an arbitrary number of character states per polymorphic site<br/>
Also equivalent to sum of site heterozygosities:<br/>
 </p>
<p class="formulaDsp">
<img class="formulaDsp" alt="\[ \widehat\theta_\pi=\sum_{i=1}^{i=S}2{p_i}{q_i} \]" src="form_23.png"/>
</p>
<p><br/>
Also equivalent to mean pairwise differences, but that's slow to calculate.<br/>
<br/>
If there is missing data (indicated by 'N' characters), the sample size is reduced for that site. For example, if the data for the <img class="formulaInl" alt="$i_{th}$" src="form_24.png"/> site is:<br/>
A<br/>
A<br/>
A<br/>
N<br/>
N<br/>
G<br/>
Then ThetaPi is calculated for that site as if the sample size were 4 (not 6), and the polymorphic site frequencies are 3/4 for A and 1/4 for G<br/>
<br/>
</p>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#aab65a802f26b230fa9d0bfd58070fe7c">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00182">182</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a942cec5aec3bd820c9029609969d4b57"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::ThetaW </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>The classic "Watterson's Theta" statistic, generalized to missing data and multiple mutations per site: </p>
<p class="formulaDsp">
<img class="formulaDsp" alt="\[ \widehat\theta_w=\sum_{i=1}^{i=S}\frac{S}{\sum_{j=1}^{j=n_i-1}\frac{1}{j}} \]" src="form_25.png"/>
</p>
<p><br/>
For this statistic, <img class="formulaInl" alt="$S$" src="form_18.png"/> is either the number of segregating sites, or the number of mutations on the genealogy and <img class="formulaInl" alt="$n_i$" src="form_22.png"/> is the sample size at site i. If totMuts == 1, the number of mutations is used, else the number ofsegregating sites is used. </p>
<dl class="section warning"><dt>Warning</dt><dd>Statistic undefined if there are untyped SNPs. In the presence of missing data, ThetaW is calculated as the sum (over all segregating sites) of 1/a_sub_n, where a_sub_n is the denominator of ThetaW, using the number of alleles without missing data as the sample size. More formally, the routine returns a calculation base on an unweighted sample size adjustment accross sites. </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a593e8e6121bcf8477a2319b5f47a27c1">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00252">252</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="aa5614fba76be5730e4618032820a9b0c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::VarPi </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Total variance of mean pairwise differences. Tajima in Takahata/Clark book, (13). </p>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00956">956</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a05276ed08aa7184738228045dd87ec15"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::VarThetaW </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>Variance of Watterson's Theta (<a class="el" href="classSequence_1_1PolySNP.html#a942cec5aec3bd820c9029609969d4b57">ThetaW()</a>). </dd></dl>
<dl class="section warning"><dt>Warning</dt><dd>statistic undefined if there are untyped SNPs </dd></dl>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l01004">1004</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a00d617041ba15a1593ffd85134a6a315"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::WallsB </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>Wall's B Statistic. Wall, J. (1999) Genetical Research 74, pp 65-79 </dd></dl>
<dl class="section author"><dt>Author</dt><dd>Kevin Thornton </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a06ce66b3dd03060754375ba534aba018">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00834">834</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="aa2360fb83235da3d1ff16eec11454b99"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">unsigned Sequence::PolySNP::WallsBprime </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>Wall's B' Statistic. Wall, J. (1999) Genetical Research 74, pp 65-79 </dd></dl>
<dl class="section author"><dt>Author</dt><dd>Kevin Thornton </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a8377b823e8358c426ad92d277f2a77ad">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00925">925</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<a class="anchor" id="a016a664f6cf9380310910d86482b1f7e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">double Sequence::PolySNP::WallsQ </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Returns</dt><dd>Wall's Q Statistic. Wall, J. (1999) Genetical Research 74, pp 65-79 </dd></dl>
<dl class="section author"><dt>Author</dt><dd>Kevin Thornton </dd></dl>

<p>Reimplemented in <a class="el" href="classSequence_1_1PolySIM.html#a737f566ec959c25729840b8f1bcf8c10">Sequence::PolySIM</a>.</p>

<p>Definition at line <a class="el" href="PolySNP_8cc_source.html#l00940">940</a> of file <a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li><a class="el" href="PolySNP_8hpp_source.html">PolySNP.hpp</a></li>
<li><a class="el" href="PolySNP_8cc_source.html">PolySNP.cc</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="namespaceSequence.html">Sequence</a></li><li class="navelem"><a class="el" href="classSequence_1_1PolySNP.html">PolySNP</a></li>
    <li class="footer">Generated on Mon Nov 9 2015 09:43:11 for libsequence by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.6 </li>
  </ul>
</div>
</body>
</html>
