<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.6"/>
<title>libsequence: Class Hierarchy</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">libsequence
   &#160;<span id="projectnumber">1.8.9</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.6 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="examples.html"><span>Examples</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li class="current"><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('hierarchy.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="headertitle">
<div class="title">Class Hierarchy</div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock">This inheritance list is sorted roughly, but not completely, alphabetically:</div><div class="directory">
<div class="levels">[detail level <span onclick="javascript:toggleLevel(1);">1</span><span onclick="javascript:toggleLevel(2);">2</span>]</div><table class="directory">
<tr id="row_0_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1__PolySNPImpl.html" target="_self">Sequence::_PolySNPImpl</a></td><td class="desc"></td></tr>
<tr id="row_1_"><td class="entry"><img id="arr_1_" src="ftv2mnode.png" alt="o" width="16" height="22" onclick="toggleFolder('1_')"/><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1AlignStream.html" target="_self">Sequence::AlignStream&lt; T &gt;</a></td><td class="desc">Virtual interface to alignment streams </td></tr>
<tr id="row_1_0_" class="even"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1ClustalW.html" target="_self">Sequence::ClustalW&lt; T &gt;</a></td><td class="desc"><a class="el" href="classSequence_1_1ClustalW.html" title="ClustalW streams. ">ClustalW</a> streams </td></tr>
<tr id="row_1_1_"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2lastnode.png" alt="\" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1phylipData.html" target="_self">Sequence::phylipData&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_2_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1ambiguousNucleotide.html" target="_self">Sequence::ambiguousNucleotide</a></td><td class="desc">Tests if a character is in the set A,G,C,T </td></tr>
<tr id="row_3_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1bamaux.html" target="_self">Sequence::bamaux</a></td><td class="desc">The extra data types at the end of a bam record </td></tr>
<tr id="row_4_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1bamreader.html" target="_self">Sequence::bamreader</a></td><td class="desc">A class managing input from BAM files </td></tr>
<tr id="row_5_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1bamreaderImpl.html" target="_self">Sequence::bamreaderImpl</a></td><td class="desc">Impl class for bamreader </td></tr>
<tr id="row_6_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1bamrecord.html" target="_self">Sequence::bamrecord</a></td><td class="desc">A single alignment record from a BAM file </td></tr>
<tr id="row_7_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1coalsim_1_1chromosome.html" target="_self">Sequence::coalsim::chromosome</a></td><td class="desc">A chromosome is a container of segments </td></tr>
<tr id="row_8_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1Comeron95.html" target="_self">Sequence::Comeron95</a></td><td class="desc">Ka and Ks by Comeron's (1995) method </td></tr>
<tr id="row_9_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1ComplementBase.html" target="_self">Sequence::ComplementBase</a></td><td class="desc"></td></tr>
<tr id="row_10_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1countDerivedStates.html" target="_self">Sequence::countDerivedStates</a></td><td class="desc">Functor to count the number of derived states, excluding gaps and missing data, in a range of characters </td></tr>
<tr id="row_11_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1countStates.html" target="_self">Sequence::countStates</a></td><td class="desc">Functor to count the number of states, excluding gaps and missing data, in a range of characters </td></tr>
<tr id="row_12_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structfirst__is__equal.html" target="_self">first_is_equal&lt; key, value &gt;</a></td><td class="desc">Functor that checks for equality of first member of two pairs </td></tr>
<tr id="row_13_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1FST.html" target="_self">Sequence::FST</a></td><td class="desc">Analysis of population structure using <img class="formulaInl" alt="$F_{ST}$" src="form_6.png"/> </td></tr>
<tr id="row_14_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1GarudStats.html" target="_self">Sequence::GarudStats</a></td><td class="desc"></td></tr>
<tr id="row_15_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1Grantham.html" target="_self">Sequence::Grantham</a></td><td class="desc"><a class="el" href="classSequence_1_1Grantham.html" title="Grantham&#39;s distances. ">Grantham</a>'s distances </td></tr>
<tr id="row_16_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1HKAdata.html" target="_self">Sequence::HKAdata</a></td><td class="desc">Data from a single locus for an HKA test </td></tr>
<tr id="row_17_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1HKAresults.html" target="_self">Sequence::HKAresults</a></td><td class="desc">Results of calculations of the HKA test </td></tr>
<tr id="row_18_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1invalidPolyChar.html" target="_self">Sequence::invalidPolyChar</a></td><td class="desc">This functor can be used to determine if a range contains characters that the SNP analysis routines in this library cannot handle gracefully </td></tr>
<tr id="row_19_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1Kimura80.html" target="_self">Sequence::Kimura80</a></td><td class="desc">Kimura's 2-parameter distance </td></tr>
<tr id="row_20_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1lowerCrit.html" target="_self">Sequence::lowerCrit</a></td><td class="desc">Find the upper critical value of a sorted list </td></tr>
<tr id="row_21_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1coalsim_1_1marginal.html" target="_self">Sequence::coalsim::marginal</a></td><td class="desc">The genealogy of a portion of a chromosome on which no recombination has occurred </td></tr>
<tr id="row_22_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1coalsim_1_1newick__stream__marginal__tree.html" target="_self">Sequence::coalsim::newick_stream_marginal_tree</a></td><td class="desc">Class that provides a typecast-on-output of a marginal tree to a newick tree Example use: </td></tr>
<tr id="row_23_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1coalsim_1_1newick__stream__marginal__tree__impl.html" target="_self">Sequence::coalsim::newick_stream_marginal_tree_impl</a></td><td class="desc"></td></tr>
<tr id="row_24_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1nmuts.html" target="_self">Sequence::nmuts&lt; counter &gt;</a></td><td class="desc">Calculate the number of mutations at a polymorphic site </td></tr>
<tr id="row_25_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1coalsim_1_1node.html" target="_self">Sequence::coalsim::node</a></td><td class="desc">A point on a marginal tree at which a coalescent event occurs </td></tr>
<tr id="row_26_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structoutputFreq.html" target="_self">outputFreq</a></td><td class="desc"></td></tr>
<tr id="row_27_"><td class="entry"><img id="arr_27_" src="ftv2mnode.png" alt="o" width="16" height="22" onclick="toggleFolder('27_')"/><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1PolySNP.html" target="_self">Sequence::PolySNP</a></td><td class="desc">Molecular population genetic analysis </td></tr>
<tr id="row_27_0_" class="even"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2lastnode.png" alt="\" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1PolySIM.html" target="_self">Sequence::PolySIM</a></td><td class="desc">Analysis of coalescent simulation data </td></tr>
<tr id="row_28_"><td class="entry"><img id="arr_28_" src="ftv2mnode.png" alt="o" width="16" height="22" onclick="toggleFolder('28_')"/><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1PolyTable.html" target="_self">Sequence::PolyTable</a></td><td class="desc">The base class for polymorphism tables </td></tr>
<tr id="row_28_0_" class="even"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1PolySites.html" target="_self">Sequence::PolySites</a></td><td class="desc">Polymorphism tables for sequence data </td></tr>
<tr id="row_28_1_"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1SimData.html" target="_self">Sequence::SimData</a></td><td class="desc">Data from coalescent simulations </td></tr>
<tr id="row_28_2_" class="even"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2lastnode.png" alt="\" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1SimpleSNP.html" target="_self">Sequence::SimpleSNP</a></td><td class="desc">SNP table data format </td></tr>
<tr id="row_29_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1PolyTableSlice.html" target="_self">Sequence::PolyTableSlice&lt; T &gt;</a></td><td class="desc">A container class for "sliding windows" along a polymorphism table </td></tr>
<tr id="row_30_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1ProductMoment.html" target="_self">Sequence::ProductMoment</a></td><td class="desc">Pearson's product-moment correlation </td></tr>
<tr id="row_31_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1RedundancyCom95.html" target="_self">Sequence::RedundancyCom95</a></td><td class="desc">Calculate redundancy of a genetic code using Comeron's counting scheme </td></tr>
<tr id="row_32_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1samflag.html" target="_self">Sequence::samflag</a></td><td class="desc">The flag field of a SAM record </td></tr>
<tr id="row_33_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1samrecord.html" target="_self">Sequence::samrecord</a></td><td class="desc">A single alignment record from a SAM file </td></tr>
<tr id="row_34_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1samtag.html" target="_self">Sequence::samtag</a></td><td class="desc"></td></tr>
<tr id="row_35_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1coalsim_1_1segment.html" target="_self">Sequence::coalsim::segment</a></td><td class="desc">A portion of a recombining chromosome </td></tr>
<tr id="row_36_" class="even"><td class="entry"><img id="arr_36_" src="ftv2mnode.png" alt="o" width="16" height="22" onclick="toggleFolder('36_')"/><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1Seq.html" target="_self">Sequence::Seq</a></td><td class="desc">Abstract interface to sequence objects </td></tr>
<tr id="row_36_0_"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1Fasta.html" target="_self">Sequence::Fasta</a></td><td class="desc">FASTA sequence stream </td></tr>
<tr id="row_36_1_" class="even"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2lastnode.png" alt="\" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1fastq.html" target="_self">Sequence::fastq</a></td><td class="desc"></td></tr>
<tr id="row_37_"><td class="entry"><img id="arr_37_" src="ftv2mnode.png" alt="o" width="16" height="22" onclick="toggleFolder('37_')"/><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1SeqException.html" target="_self">Sequence::SeqException</a></td><td class="desc">Base class for exceptions that may be thrown </td></tr>
<tr id="row_37_0_" class="even"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2lastnode.png" alt="\" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1badFormat.html" target="_self">Sequence::badFormat</a></td><td class="desc"></td></tr>
<tr id="row_38_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1shortestPath.html" target="_self">Sequence::shortestPath</a></td><td class="desc">Calculate shortest path between 2 codons </td></tr>
<tr id="row_39_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1SimParams.html" target="_self">Sequence::SimParams</a></td><td class="desc">Parameters for Hudson's simulation program </td></tr>
<tr id="row_40_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1SingleSub.html" target="_self">Sequence::SingleSub</a></td><td class="desc">Deal with codons differing at 1 position </td></tr>
<tr id="row_41_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1Sites.html" target="_self">Sequence::Sites</a></td><td class="desc">Calculate length statistics for divergence calculations </td></tr>
<tr id="row_42_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1SpearmansRank.html" target="_self">Sequence::SpearmansRank</a></td><td class="desc">Spearman's rank correlation </td></tr>
<tr id="row_43_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1ssh.html" target="_self">Sequence::ssh</a></td><td class="desc">Calculate nucleotide diversity from a polymorphic site </td></tr>
<tr id="row_44_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1stateCounter.html" target="_self">Sequence::stateCounter</a></td><td class="desc">Keep track of state counts at a site in an alignment or along a sequence </td></tr>
<tr id="row_45_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1Sums.html" target="_self">Sequence::Sums&lt; T &gt;</a></td><td class="desc"></td></tr>
<tr id="row_46_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1ThreeSubs.html" target="_self">Sequence::ThreeSubs</a></td><td class="desc">Deal with codons differing at all 3 positions </td></tr>
<tr id="row_47_" class="even"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="classSequence_1_1TwoSubs.html" target="_self">Sequence::TwoSubs</a></td><td class="desc">Deal with codons differing at 2 positions </td></tr>
<tr id="row_48_"><td class="entry"><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1upperCrit.html" target="_self">Sequence::upperCrit</a></td><td class="desc">Find the upper critical value of a sorted list </td></tr>
<tr id="row_49_" class="even"><td class="entry"><img id="arr_49_" src="ftv2mnode.png" alt="o" width="16" height="22" onclick="toggleFolder('49_')"/><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1WeightingScheme2.html" target="_self">Sequence::WeightingScheme2</a></td><td class="desc">Abstract interface to weighting schemes when codons differ at 2 positions </td></tr>
<tr id="row_49_0_"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1GranthamWeights2.html" target="_self">Sequence::GranthamWeights2</a></td><td class="desc">Weights paths by <a class="el" href="classSequence_1_1Grantham.html" title="Grantham&#39;s distances. ">Grantham</a>'s distances for codons differing at 2 sites </td></tr>
<tr id="row_49_1_" class="even"><td class="entry"><img src="ftv2vertline.png" alt="|" width="16" height="22" /><img src="ftv2lastnode.png" alt="\" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1Unweighted2.html" target="_self">Sequence::Unweighted2</a></td><td class="desc">Weights all pathways equally </td></tr>
<tr id="row_50_"><td class="entry"><img id="arr_50_" src="ftv2mlastnode.png" alt="\" width="16" height="22" onclick="toggleFolder('50_')"/><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1WeightingScheme3.html" target="_self">Sequence::WeightingScheme3</a></td><td class="desc">Abstract interface to weighting schemes when codons differ at 3 positions </td></tr>
<tr id="row_50_0_" class="even"><td class="entry"><img src="ftv2blank.png" alt="&#160;" width="16" height="22" /><img src="ftv2node.png" alt="o" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1GranthamWeights3.html" target="_self">Sequence::GranthamWeights3</a></td><td class="desc">Weights paths by <a class="el" href="classSequence_1_1Grantham.html" title="Grantham&#39;s distances. ">Grantham</a>'s distances for codons differing at 3 sites </td></tr>
<tr id="row_50_1_"><td class="entry"><img src="ftv2blank.png" alt="&#160;" width="16" height="22" /><img src="ftv2lastnode.png" alt="\" width="16" height="22" /><img src="ftv2cl.png" alt="C" width="24" height="22" /><a class="el" href="structSequence_1_1Unweighted3.html" target="_self">Sequence::Unweighted3</a></td><td class="desc">Weights all pathways equally </td></tr>
</table>
</div><!-- directory -->
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Mon Feb 15 2016 14:45:54 for libsequence by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.6 </li>
  </ul>
</div>
</body>
</html>
